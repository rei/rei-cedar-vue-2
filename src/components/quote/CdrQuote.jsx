import clsx from 'clsx';
import CdrText from '../text/CdrText';
import modifier from '../../mixins/modifier';
import style from './styles/CdrQuote.scss';

export default {
  name: 'CdrQuote',
  components: {
    CdrText,
  },
  mixins: [modifier],
  props: {
    tag: {
      type: String,
      default: 'blockquote',
      validator: (value) => (['blockquote', 'aside', 'q', 'div'].indexOf(value) >= 0) || false,

    },
    /** Caption summary text */
    summary: String,
    /** Caption credit text */
    citation: String,
  },
  data() {
    return {
      style,
    };
  },
  computed: {
    baseClass() {
      return 'cdr-quote';
    },
    summaryBlock() {
      return this.summary ? (
        <p
          class={this.style['cdr-quote__summary']}
        >
          { this.summary }
        </p>
      ) : '';
    },
    citationBlock() {
      return this.citation ? (
        <cdr-text
          tag="cite"
          class={this.style['cdr-quote__citation']}
        >
          { this.citation }
        </cdr-text>
      ) : '';
    },
  },
  render() {
    const Component = this.tag;
    return (<Component
      class={clsx(this.style[this.baseClass], this.modifierClass)}
    >
      {this.summaryBlock}
      {this.$slots.default}
      {this.citationBlock}
    </Component>);
  },
};
