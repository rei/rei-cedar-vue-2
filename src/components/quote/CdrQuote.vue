<template>
  <component
    :is="tag"
    :class="[modifierClass]"
  >
    <cdr-text
      v-if="summary"
      class="cdr-quote__summary">
      {{ summary }}
    </cdr-text>
    <slot />
    <cdr-text
      tag="cite"
      v-if="citation"
      modifier="citation">
      {{ citation }}
    </cdr-text>
  </component>
</template>

<script>
import { CdrText } from '@rei/cdr-text';
import modifier from 'mixinsdir/modifier';

/**
 * Cedar 2 component for captions
 * @version 0.0.1
 * @author [REI Software Engineering](https://rei.github.io/rei-cedar/)
 */
export default {
  name: 'CdrQuote',
  components: {
    CdrText,
  },
  mixins: [modifier],
  props: {
    tag: {
      type: String,
      default: 'blockquote',
      validator: value => (['blockquote', 'aside', 'q', 'div'].indexOf(value) >= 0) || false,

    },
    /** Caption summary text */
    summary: String,
    /** Caption credit text */
    citation: String,
  },
  computed: {
    baseClass() {
      return 'cdr-quote';
    },
  },
};
</script>

<style>
  @import '../../css/settings/_index.pcss';
  @import './styles/spruceQuote.min.css';
  @import './styles/CdrQuote.pcss';
</style>
