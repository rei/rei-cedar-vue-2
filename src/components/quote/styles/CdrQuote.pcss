/* ==========================================================================
  # Cdrquote
  
  TOC:

    :Base - Cdrquote
========================================================================== */
@import './spruceQuote.css';

.cdr-quote {
  margin: 0;
  padding: $inset-1-x-stretch;
  
  &__summary {
    font-size: 20px;
    font-style: italic;
    font-family: Sentinel, Roboto, "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-weight: 400;
    line-height: 28px;
    letter-spacing: 0;
    margin: 0;

    @media (--sm-mq) {
      font-size: 24px;
      line-height: 32px;
    }

    & + .cdr-quote__citation {
      display: block;
      padding-top: $space-1-x;
    }
  }

  &--pull {
    border-style: solid;
    border-color: #222;

    .cdr-quote__summary {
      font-weight: 600;
    }

    @media (--xs-mq) {
      border-width: 0 0 1px 0;
      padding: $inset-1-x-stretch;
      margin: 0 0 $space-1-x;
    }

    @media (--sm-mq) {
      border-width: 0 0 0 1px;
      padding: $space-half-x $space-1-x $space-half-x $space-2-x;
      margin: $space-1-x 0;
    }
  }
}
