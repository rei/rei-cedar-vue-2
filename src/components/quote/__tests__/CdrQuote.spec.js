import { mount } from '@vue/test-utils';
import CdrQuote from 'componentdir/quote/CdrQuote';

describe('CdrQuote', () => {
  it('renders correctly', () => {
    const wrapper = mount(CdrQuote, {
      propsData: {
        tag: 'aside',
        summary: 'This is the summary',
        citation: 'This is the citation'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
