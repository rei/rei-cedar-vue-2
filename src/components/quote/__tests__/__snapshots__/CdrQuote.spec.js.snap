// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrQuote renders correctly 1`] = `
<aside
  class="cdr-quote"
>
  <p
    class="cdr-quote__summary"
  >
    This is the summary
  </p>
  <cite
    class="cdr-text cdr-text--citation cdr-quote__citation"
  >
    This is the citation
  </cite>
</aside>
`;
